inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -171,7 +171,7 @@ def _rollback_and_delete_snap(self) -> None:
         a.) After reboot by SOTA, and diagnostic reports bad report for system health
         """
         logger.debug("")
-        if self.snap_num:
+        if self.snap_num and self.snap_num is not "0":
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Maintainability$</h3>

Add a comment explaining why the snapshot number "0" is treated as a special case, for future maintainability.